### PR TITLE
Change everything to relative urls to fix proxies

### PR DESCRIPTION
**Pull request by voidus**
http://jira.opensource.mayflower.de/jira/browse/PHPROJEKT-317

### CODE REVIEW

File: phprojekt/application/Contact/Views/dojo/scripts/Form.js [hunk 1]
```diff
@@ -62,7 +62,7 @@ dojo.declare("phpr.Contact.Form", phpr.Default.DialogForm, {
 
     deleteForm: function() {
         phpr.send({
-            url:       phpr.webpath + 'index.php/' + phpr.module + '/index/jsonDelete/id/' + this.id
+            url:       'index.php/' + phpr.module + '/index/jsonDelete/id/' + this.id
```
Comment (exi) [inline phprojekt/application/Contact/Views/dojo/scripts/Form.js]: whitespaces

File: phprojekt/application/Core/Views/dojo/scripts/Administration/Module/Form.js [hunk 2]
```diff
@@ -51,7 +51,7 @@ dojo.declare("phpr.Module.Form", phpr.Core.DialogForm, {
 
     initData: function() {
         // Get all the active users
-        this._moduleDesignerUrl  = phpr.webpath + 'index.php/Core/moduleDesigner/jsonDetail/nodeId/1/id/' + this.id;
+        this._moduleDesignerUrl  = 'index.php/Core/moduleDesigner/jsonDetail/nodeId/1/id/' + this.id;
```
Comment (exi) [inline phprojekt/application/Core/Views/dojo/scripts/Administration/Module/Form.js]: whitespaces

File: phprojekt/application/Core/Views/dojo/scripts/Administration/Module/Form.js [hunk 3]
```diff
@@ -291,14 +290,14 @@ dojo.declare("phpr.Module.Form", phpr.Core.DialogForm, {
 
         this.setSubmitInProgress(true);
         phpr.send({
-            url:       phpr.webpath + 'index.php/Core/moduleDesigner/jsonSave/nodeId/1/id/' + this.id,
+            url:       'index.php/Core/moduleDesigner/jsonSave/nodeId/1/id/' + this.id,
```
Comment (exi) [inline phprojekt/application/Core/Views/dojo/scripts/Administration/Module/Form.js]: whitespaces

File: phprojekt/application/Core/Views/dojo/scripts/Administration/Module/Form.js [hunk 4]
```diff
@@ -323,7 +322,7 @@ dojo.declare("phpr.Module.Form", phpr.Core.DialogForm, {
 
     deleteForm: function() {
         phpr.send({
-            url:       phpr.webpath + 'index.php/Core/' + phpr.module.toLowerCase() + '/jsonDelete/id/' + this.id
+            url:       'index.php/Core/' + phpr.module.toLowerCase() + '/jsonDelete/id/' + this.id
```
Comment (exi) [inline phprojekt/application/Core/Views/dojo/scripts/Administration/Module/Form.js]: whitespaces

File: phprojekt/application/Default/Views/dojo/scripts/Form.js [hunk 5]
```diff
@@ -1075,13 +1075,13 @@ dojo.declare("phpr.Default.Form", phpr.Default.System.Component, {
         var pid = phpr.currentProjectId;
 
         phpr.send({
-            url:       phpr.webpath + 'index.php/' + phpr.module + '/index/jsonDelete/id/' + this.id
+            url:       'index.php/' + phpr.module + '/index/jsonDelete/id/' + this.id
```
Comment (exi) [inline phprojekt/application/Default/Views/dojo/scripts/Form.js:1077]: whitespaces

File: phprojekt/application/Default/Views/dojo/scripts/Grid.js [hunk 6]
```diff
@@ -169,7 +169,7 @@ dojo.declare("phpr.Default.Grid", phpr.Default.System.Component, {
         // Description:
         //    Draw the tags
         // Get the module tags
-        this._tagUrl  = phpr.webpath + 'index.php/Default/Tag/jsonGetTags';
+        this._tagUrl  = 'index.php/Default/Tag/jsonGetTags';
```
Comment (exi) [inline phprojekt/application/Default/Views/dojo/scripts/Grid.js]: whitespaces

File: phprojekt/application/Gantt/Views/dojo/scripts/Main.js [hunk 7]
```diff
@@ -450,7 +448,7 @@ dojo.declare("phpr.Gantt.Main", phpr.Default.Main, {
         }
         sendData['projects[]'] = projects;
         phpr.send({
-            url:       phpr.webpath + 'index.php/Gantt/index/jsonSave/nodeId/' + phpr.currentProjectId,
+            url:       'index.php/Gantt/index/jsonSave/nodeId/' + phpr.currentProjectId,
             content:   sendData
```
Comment (exi) [inline phprojekt/application/Gantt/Views/dojo/scripts/Main.js]: whitespaces

File: phprojekt/application/Project/Views/dojo/scripts/FormBasicData.js [hunk 8]
```diff
@@ -47,7 +47,7 @@ dojo.declare("phpr.Project.FormBasicData", phpr.Project.Form, {
         this._initData.push({'store': this.moduleStore});
 
         // Get the tags
-        this._tagUrl  = phpr.webpath + 'index.php/Default/Tag/jsonGetTagsByModule/moduleName/' + phpr.module +
+        this._tagUrl  = 'index.php/Default/Tag/jsonGetTagsByModule/moduleName/' + phpr.module +
```
Comment (exi) [inline phprojekt/application/Project/Views/dojo/scripts/FormBasicData.js]: whitespace

File: phprojekt/library/Phprojekt.php [hunk 9]
```diff
@@ -447,19 +447,7 @@ public function _initialize()
...
-                . $response->getBasePath() . '/';
-            $this->_config->basepath = $response->getBasePath() . '/';
-        }
-        if (!defined('PHPR_ROOT_WEB_PATH')) {
-            define('PHPR_ROOT_WEB_PATH', $this->_config->webpath . 'index.php/');
-        }
-        if (!defined('PHPR_ROOT_WEB_BASE_PATH')) {
-            define('PHPR_ROOT_WEB_BASE_PATH', $this->_config->basepath . 'index.php/');
-        }
+        // Set tmpPath and applicationPath
```
Comment (exi) [inline phprojekt/library/Phprojekt.php]: pointless comment

File: phprojekt/library/Phprojekt/Notification.php [hunk 10]
```diff
@@ -277,7 +277,9 @@ public function getBodyParams()
         $bodyParams['moduleTable'] = $this->_model->getModelName();
 
         // Url
-        $url = Phprojekt::getInstance()->getConfig()->webpath . "index.php#" . $this->_model->getModelName();
+        $request = new Zend_Controller_Request_Http();
+        $bodyParams['url'] = $request->getScheme() . '://' . $request->getHttpHost()
+                . $request->getBasePath() . '/index.php#' . $this->_model->getModelName();
```
Comment (exi) [inline phprojekt/library/Phprojekt/Notification.php]: why would we replace a configurable webpath url from the config, which is proxy compatible, with the url from the request object, which is not?

File: phprojekt/application/Calendar2/Models/Notification.php [hunk 11]
```diff
@@ -79,7 +79,9 @@ public function getBodyParams()
         if ($this->_model->recurrenceId) {
             $params['occurrence'] = $this->_model->recurrenceId;
         }
-        $bodyParams['url'] = Phprojekt::getInstance()->getConfig()->webpath . 'index.php#' . http_build_query($params);
+        $request = new Zend_Controller_Request_Http();
+        $bodyParams['url'] = $request->getScheme() . '://' . $request->getHttpHost()
+                . $request->getBasePath() . '/index.php#' . http_build_query($params);
```
Comment (exi) [inline phprojekt/application/Calendar2/Models/Notification.php]: why would we replace a configurable webpath url from the config, which is proxy compatible, with the url from the request object, which is not?

File: phprojekt/library/Phprojekt/Auth.php [hunk 12]
```diff
@@ -688,7 +688,8 @@ private static function _saveLoginData($userId)
     private static function _setCookies($hash, $userId, $expires)
     {
         // Set cookies
-        $completePath     = Phprojekt::getInstance()->getConfig()->webpath;
+        $request          = new Zend_Controller_Request_Http();
+        $completePath     = $request->getScheme() . '://' . $request->getHttpHost() . $request->getBasePath() . '/';
```
Comment (exi) [inline phprojekt/library/Phprojekt/Auth.php]: why would we replace a configurable webpath url from the config, which is proxy compatible, with the url from the request object, which is not?

File: phprojekt/application/Default/Views/dojo/scripts/Form.js [hunk 13]
```diff
@@ -174,12 +174,12 @@ dojo.declare("phpr.Default.Form", phpr.Default.System.Component, {
...
 
         // Get the rights for other users
-        this._accessUrl = phpr.webpath + 'index.php/' + phpr.module +
+        this._accessUrl = 'index.php/' + phpr.module +
             '/index/jsonGetUsersRights' + '/nodeId/' + phpr.currentProjectId + '/id/' + this.id;
         this._initData.push({'url': this._accessUrl});
 
         // Get the tags
-        this._tagUrl  = phpr.webpath + 'index.php/Default/Tag/jsonGetTagsByModule/moduleName/' +
+        this._tagUrl  = 'index.php/Default/Tag/jsonGetTagsByModule/moduleName/' +
```
Comment (exi) [inline phprojekt/application/Default/Views/dojo/scripts/Form.js]: one space too much

Comment (voidus) [inline phprojekt/application/Default/Views/dojo/scripts/Form.js]: fixed

**Pull request merged**